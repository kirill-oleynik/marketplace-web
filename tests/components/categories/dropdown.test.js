const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const {
  CategoriesDropdown
} = require('../../../app/components/categories/dropdown');

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      <CategoriesDropdown
        t={(translation) => translation}
        categories={[
          { id: 1, title: 'Test1' },
          { id: 2, title: 'Test2' }
        ]}
      />
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#toggle', () => {
  test('it changes isOpen state', () => {
    const component = shallow(
      <CategoriesDropdown
        t={(translation) => translation}
        categories={[
          { id: 1, title: 'Test1' },
          { id: 2, title: 'Test2' }
        ]}
      />
    );

    expect(component.state().isOpen).toEqual(false);

    component.instance().toggle();

    expect(component.state().isOpen).toEqual(true);
  });
});
