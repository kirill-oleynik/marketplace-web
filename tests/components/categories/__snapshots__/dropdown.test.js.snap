// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Dropdown
  className="main-dropdown categories__dropdown"
  isOpen={false}
  tag="div"
  toggle={[Function]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    className="main-dropdown__toggle categories__dropdown-btn categories__dropdown-btn--"
    color="secondary"
    data-toggle="dropdown"
  >
    categories
  </DropdownToggle>
  <DropdownMenu
    className="categories__dropdown-menu"
    tag="div"
  >
    <a
      className="categories-link"
      href="/"
    >
      Test1
    </a>
    <a
      className="categories-link"
      href="/"
    >
      Test2
    </a>
  </DropdownMenu>
</Dropdown>
`;
